modin/engines/base/frame/partition.py

```diff
@@ -136,17 +140,55 @@ def mask(self, row_indices, col_indices):
 
         Parameters
         ----------
-        row_indices : list-like
+        row_indices : list-like, slice or label
             The indices for the rows to extract.
-        col_indices : list-like
+        col_indices : list-like, slice or label
             The indices for the columns to extract.
 
         Returns
         -------
         PandasFramePartition
             New `PandasFramePartition` object.
         """
-        pass
+
+        def is_full_axis_mask(index, axis_length):
+            """Check whether `index` mask grabs `axis_length` amount of elements."""
+            if isinstance(index, slice):
+                return index == slice(None) or (
+                    isinstance(axis_length, int)
+                    and compute_sliced_len(index, axis_length) == axis_length
+                )
+            return (
+                hasattr(index, "__len__")
+                and isinstance(axis_length, int)
+                and len(index) == axis_length
+            )
+
+        row_indices = [row_indices] if is_scalar(row_indices) else row_indices
+        col_indices = [col_indices] if is_scalar(col_indices) else col_indices
+
+        if is_full_axis_mask(row_indices, self._length_cache) and is_full_axis_mask(
```

Comment:
shouldn't we swap this check and the change to the list in the lines above?..

Reply:
I think no, otherwise we need to also add a scalar check to the `is_full_axis_mask`, which is redundant since we can just convert scalar values to the list first.